From 29adc6b0465e2d8869e65f05c685d488584a9544 Mon Sep 17 00:00:00 2001
From: Andrei Botila <andrei.botila@nxp.com>
Date: Wed, 23 Feb 2022 18:19:31 +0200
Subject: [PATCH 003/103] s32: remove s32-gen1-regs header

Remove s32-gen1-regs.h and move the specific defines
to coresponding files.

Issue: ALB-8525
Upstream-Status: Pending 

Signed-off-by: Andrei Botila <andrei.botila@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h |   3 +-
 .../asm/arch-s32/s32-gen1/s32-gen1-regs.h     | 204 ------------------
 drivers/i2c/mxc_i2c.c                         |  20 +-
 drivers/pci/serdes_s32gen1.h                  |   4 +-
 4 files changed, 10 insertions(+), 221 deletions(-)
 delete mode 100644 arch/arm/include/asm/arch-s32/s32-gen1/s32-gen1-regs.h

diff --git a/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h b/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h
index 0b4b6418bb..5b07d75ba2 100644
--- a/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h
+++ b/arch/arm/cpu/armv8/s32/s32-gen1/include/cmu.h
@@ -7,7 +7,6 @@
 #define __S32_CMU_H
 
 #include <common.h>
-#include "s32-gen1-regs.h"
 
 #ifndef __STR
 #define __STR(x) #x
@@ -24,6 +23,8 @@
 #define FXOSC_FREQ			((double)40)
 #define SIRC_FREQ			((double)0.032)
 
+#define	CMU_BASE_ADDR			(0x4005C000)
+
 #define CMU(ID, REF_CLK, MON_CLK, REF_FRQ, EXP_FRQ, REF_VAR, MON_VAR, FC) \
 {\
 	.addr = CMU_BASE_ADDR + 0x20 * (ID),\
diff --git a/arch/arm/include/asm/arch-s32/s32-gen1/s32-gen1-regs.h b/arch/arm/include/asm/arch-s32/s32-gen1/s32-gen1-regs.h
deleted file mode 100644
index a31b831183..0000000000
--- a/arch/arm/include/asm/arch-s32/s32-gen1/s32-gen1-regs.h
+++ /dev/null
@@ -1,204 +0,0 @@
-/* SPDX-License-Identifier: GPL-2.0+ */
-/*
- * Copyright 2017-2022 NXP
- */
-
-#ifndef __ASM_ARCH_S32_GEN1_REGS_H__
-#define __ASM_ARCH_S32_GEN1_REGS_H__
-
-#include <config.h>
-
-#if !defined(__ASSEMBLER__)
-#include <inttypes.h>
-
-#define UPTR(PTR)				((uintptr_t)(PTR))
-
-#define PER_GROUP0_BASE				(0x40000000UL)
-#define PER_GROUP1_BASE				(0x40100000UL)
-#define PER_GROUP2_BASE				(0x40200000UL)
-#define PER_GROUP3_BASE				(0x40300000UL)
-#define PER_GROUP8_BASE				(0x44000000UL)
-#else
-#define PER_GROUP0_BASE				(0x40000000)
-#define PER_GROUP1_BASE				(0x40100000)
-#define PER_GROUP2_BASE				(0x40200000)
-#define PER_GROUP3_BASE				(0x40300000)
-#define PER_GROUP8_BASE				(0x44000000)
-#endif
-
-/* Peripheral group 0 */
-#define POST_BASE_ADDR				(PER_GROUP0_BASE)
-#define MTR_BASE_ADDR				(PER_GROUP0_BASE + 0x0004000)
-#define ATX_BASE_ADDR				(PER_GROUP0_BASE + 0x0008000)
-#define STAM_BASE_ADDR				(PER_GROUP0_BASE + 0x000C000)
-#define TST_CLK_MUX_BASE_ADDR			(PER_GROUP0_BASE + 0x0010000)
-#define ADC_BIST0_BASE_ADDR			(PER_GROUP0_BASE + 0x0014000)
-#define ADC_BIST1_BASE_ADDR			(PER_GROUP0_BASE + 0x0018000)
-#define STCU_BASE_ADDR				(PER_GROUP0_BASE + 0x0028000)
-
-#define MC_CGM0_BASE_ADDR			(PER_GROUP0_BASE + 0x0030000)
-#define MC_CGM1_BASE_ADDR			(PER_GROUP0_BASE + 0x0034000)
-#define ARM_PLL_BASE_ADDR			(PER_GROUP0_BASE + 0x0038000)
-#define PERIPH_PLL_BASE_ADDR			(PER_GROUP0_BASE + 0x003C000)
-#define ACCEL_PLL_BASE_ADDR			(PER_GROUP0_BASE + 0x0040000)
-#define DRAM_PLL_BASE_ADDR			(PER_GROUP0_BASE + 0x0044000)
-#define DEBUG_MUX_BASE_ADDR			(PER_GROUP0_BASE + 0x0048000)
-#define XOSC_BASE_ADDR				(PER_GROUP0_BASE + 0x0050000)
-#define ARM_DFS_BASE_ADDR			(PER_GROUP0_BASE + 0x0054000)
-#define PERIPH_DFS_BASE_ADDR			(PER_GROUP0_BASE + 0x0058000)
-#define CMU_BASE_ADDR				(PER_GROUP0_BASE + 0x005C000)
-#define RTC_BASE_ADDR				(PER_GROUP0_BASE + 0x0060000)
-#define MC_CGM5_BASE_ADDR			(PER_GROUP0_BASE + 0x0068000)
-#define MC_RGM_BASE_ADDR			(PER_GROUP0_BASE + 0x0078000)
-#define SRC_SOC_BASE_ADDR			(PER_GROUP0_BASE + 0x007C000)
-#define RDC_BASE_ADDR				(PER_GROUP0_BASE + 0x0080000)
-#define MC_ME_BASE_ADDR				(PER_GROUP0_BASE + 0x0088000)
-#define PMC_BASE_ADDR				(PER_GROUP0_BASE + 0x008C000)
-#define WKPU_BASE_ADDR				(PER_GROUP0_BASE + 0x0090000)
-#define SIUL2_0_BASE_ADDR			(PER_GROUP0_BASE + 0x009C000)
-#define TMU_BASE_ADDR				(PER_GROUP0_BASE + 0x00A8000)
-
-/* Peripheral group 1 */
-#define SWT0_BASE_ADDR				(PER_GROUP1_BASE)
-#define SWT1_BASE_ADDR				(PER_GROUP1_BASE + 0x00004000)
-#define SWT2_BASE_ADDR				(PER_GROUP1_BASE + 0x00008000)
-#define SWT3_BASE_ADDR				(PER_GROUP1_BASE + 0x0000C000)
-#define STM0_BASE_ADDR				(PER_GROUP1_BASE + 0x0001C000)
-#define STM1_BASE_ADDR				(PER_GROUP1_BASE + 0x00020000)
-#define STM2_BASE_ADDR				(PER_GROUP1_BASE + 0x00024000)
-#define STM3_BASE_ADDR				(PER_GROUP1_BASE + 0x00028000)
-#define DMAMUX0_BASE_ADDR			(PER_GROUP1_BASE + 0x0002C000)
-#define DMAMUX1_BASE_ADDR			(PER_GROUP1_BASE + 0x00030000)
-#define DMACR0_BASE_ADDR			(PER_GROUP1_BASE + 0x0003C000)
-#define EDMA0_CONTROL_BASE_ADDR			(PER_GROUP1_BASE + 0x00044000)
-#define EDMA0_CHANNEL_0_BASE_ADDR		(PER_GROUP1_BASE + 0x00048000)
-#define EDMA0_CHANNEL_1_BASE_ADDR		(PER_GROUP1_BASE + 0x00049000)
-#define EDMA0_CHANNEL_2_BASE_ADDR		(PER_GROUP1_BASE + 0x0004A000)
-#define EDMA0_CHANNEL_3_BASE_ADDR		(PER_GROUP1_BASE + 0x0004B000)
-#define EDMA0_CHANNEL_4_BASE_ADDR		(PER_GROUP1_BASE + 0x0004C000)
-#define EDMA0_CHANNEL_5_BASE_ADDR		(PER_GROUP1_BASE + 0x0004D000)
-#define EDMA0_CHANNEL_6_BASE_ADDR		(PER_GROUP1_BASE + 0x0004E000)
-#define EDMA0_CHANNEL_7_BASE_ADDR		(PER_GROUP1_BASE + 0x0004F000)
-#define EDMA0_CHANNEL_8_BASE_ADDR		(PER_GROUP1_BASE + 0x00050000)
-#define EDMA0_CHANNEL_9_BASE_ADDR		(PER_GROUP1_BASE + 0x00051000)
-#define EDMA0_CHANNEL_10_BASE_ADDR		(PER_GROUP1_BASE + 0x00052000)
-#define EDMA0_CHANNEL_11_BASE_ADDR		(PER_GROUP1_BASE + 0x00053000)
-#define EDMA0_CHANNEL_12_BASE_ADDR		(PER_GROUP1_BASE + 0x00054000)
-#define EDMA0_CHANNEL_13_BASE_ADDR		(PER_GROUP1_BASE + 0x00055000)
-#define EDMA0_CHANNEL_14_BASE_ADDR		(PER_GROUP1_BASE + 0x00056000)
-#define EDMA0_CHANNEL_15_BASE_ADDR		(PER_GROUP1_BASE + 0x00057000)
-#define EDMA0_CHANNEL_16_BASE_ADDR		(PER_GROUP1_BASE + 0x00058000)
-#define EDMA0_CHANNEL_17_BASE_ADDR		(PER_GROUP1_BASE + 0x00059000)
-#define EDMA0_CHANNEL_18_BASE_ADDR		(PER_GROUP1_BASE + 0x0005A000)
-#define EDMA0_CHANNEL_19_BASE_ADDR		(PER_GROUP1_BASE + 0x0005B000)
-#define EDMA0_CHANNEL_20_BASE_ADDR		(PER_GROUP1_BASE + 0x0005C000)
-#define EDMA0_CHANNEL_21_BASE_ADDR		(PER_GROUP1_BASE + 0x0005D000)
-#define EDMA0_CHANNEL_22_BASE_ADDR		(PER_GROUP1_BASE + 0x0005E000)
-#define EDMA0_CHANNEL_23_BASE_ADDR		(PER_GROUP1_BASE + 0x0005F000)
-#define EDMA0_CHANNEL_24_BASE_ADDR		(PER_GROUP1_BASE + 0x00060000)
-#define EDMA0_CHANNEL_25_BASE_ADDR		(PER_GROUP1_BASE + 0x00061000)
-#define EDMA0_CHANNEL_26_BASE_ADDR		(PER_GROUP1_BASE + 0x00062000)
-#define EDMA0_CHANNEL_27_BASE_ADDR		(PER_GROUP1_BASE + 0x00063000)
-#define EDMA0_CHANNEL_28_BASE_ADDR		(PER_GROUP1_BASE + 0x00064000)
-#define EDMA0_CHANNEL_29_BASE_ADDR		(PER_GROUP1_BASE + 0x00065000)
-#define EDMA0_CHANNEL_30_BASE_ADDR		(PER_GROUP1_BASE + 0x00066000)
-#define EDMA0_CHANNEL_31_BASE_ADDR		(PER_GROUP1_BASE + 0x00067000)
-#define PIT0_BASE_ADDR				(PER_GROUP1_BASE + 0x00088000)
-#define CRC0_BASE_ADDR				(PER_GROUP1_BASE + 0x00090000)
-#define SRAM_CTL_BASE_ADDR			(PER_GROUP1_BASE + 0x0009C000)
-#define XRDC_BASE_ADDR				(PER_GROUP1_BASE + 0x000A4000)
-#ifdef CONFIG_NXP_S32R45
-#define CAN_FD0_BASE_ADDR			(PER_GROUP1_BASE + 0x000B4000)
-#define CAN_FD1_BASE_ADDR			(PER_GROUP1_BASE + 0x000BE000)
-#elif defined(CONFIG_NXP_S32G2XX)
-#define CAN_FD0_BASE_ADDR			(PER_GROUP1_BASE + 0x000B8000)
-#define CAN_FD1_BASE_ADDR			(PER_GROUP1_BASE + 0x000BC000)
-#endif
-#define LINFLEXD0_BASE_ADDR			(PER_GROUP1_BASE + 0x000C8000)
-#define LINFLEXD1_BASE_ADDR			(PER_GROUP1_BASE + 0x000CC000)
-#define I2C1_BASE_ADDR				(PER_GROUP1_BASE + 0x000E4000)
-#define I2C2_BASE_ADDR				(PER_GROUP1_BASE + 0x000E8000)
-#define I2C3_BASE_ADDR				(PER_GROUP1_BASE + 0x000EC000)
-#define FLEXTIMER0_BASE_ADDR			(PER_GROUP1_BASE + 0x000F4000)
-#define SARADC0_BASE_ADDR			(PER_GROUP1_BASE + 0x000F8000)
-#define CTU_BASE_ADDR				(PER_GROUP1_BASE + 0x000FC000)
-
-/* Peripheral group 2 */
-#define SWT4_BASE_ADDR				(PER_GROUP2_BASE)
-#define SWT5_BASE_ADDR				(PER_GROUP2_BASE + 0x00004000)
-#define SWT6_BASE_ADDR				(PER_GROUP2_BASE + 0x00008000)
-#define SWT7_BASE_ADDR				(PER_GROUP2_BASE + 0x0000C000)
-#define STM4_BASE_ADDR				(PER_GROUP2_BASE + 0x0001C000)
-#define STM5_BASE_ADDR				(PER_GROUP2_BASE + 0x00020000)
-#define STM6_BASE_ADDR				(PER_GROUP2_BASE + 0x00024000)
-#define STM7_BASE_ADDR				(PER_GROUP2_BASE + 0x00028000)
-#define DMAMUX2_BASE_ADDR			(PER_GROUP2_BASE + 0x0002C000)
-#define DMAMUX3_BASE_ADDR			(PER_GROUP2_BASE + 0x00030000)
-#define DMACR1_BASE_ADDR			(PER_GROUP2_BASE + 0x0003C000)
-#define EDMA1_CONTROL_BASE_ADDR			(PER_GROUP2_BASE + 0x00044000)
-#define EDMA1_CHANNEL_0_BASE_ADDR		(PER_GROUP2_BASE + 0x00048000)
-#define EDMA1_CHANNEL_1_BASE_ADDR		(PER_GROUP2_BASE + 0x00049000)
-#define EDMA1_CHANNEL_2_BASE_ADDR		(PER_GROUP2_BASE + 0x0004A000)
-#define EDMA1_CHANNEL_3_BASE_ADDR		(PER_GROUP2_BASE + 0x0004B000)
-#define EDMA1_CHANNEL_4_BASE_ADDR		(PER_GROUP2_BASE + 0x0004C000)
-#define EDMA1_CHANNEL_5_BASE_ADDR		(PER_GROUP2_BASE + 0x0004D000)
-#define EDMA1_CHANNEL_6_BASE_ADDR		(PER_GROUP2_BASE + 0x0004E000)
-#define EDMA1_CHANNEL_7_BASE_ADDR		(PER_GROUP2_BASE + 0x0004F000)
-#define EDMA1_CHANNEL_8_BASE_ADDR		(PER_GROUP2_BASE + 0x00050000)
-#define EDMA1_CHANNEL_9_BASE_ADDR		(PER_GROUP2_BASE + 0x00051000)
-#define EDMA1_CHANNEL_10_BASE_ADDR		(PER_GROUP2_BASE + 0x00052000)
-#define EDMA1_CHANNEL_11_BASE_ADDR		(PER_GROUP2_BASE + 0x00053000)
-#define EDMA1_CHANNEL_12_BASE_ADDR		(PER_GROUP2_BASE + 0x00054000)
-#define EDMA1_CHANNEL_13_BASE_ADDR		(PER_GROUP2_BASE + 0x00055000)
-#define EDMA1_CHANNEL_14_BASE_ADDR		(PER_GROUP2_BASE + 0x00056000)
-#define EDMA1_CHANNEL_15_BASE_ADDR		(PER_GROUP2_BASE + 0x00057000)
-#define EDMA1_CHANNEL_16_BASE_ADDR		(PER_GROUP2_BASE + 0x00058000)
-#define EDMA1_CHANNEL_17_BASE_ADDR		(PER_GROUP2_BASE + 0x00059000)
-#define EDMA1_CHANNEL_18_BASE_ADDR		(PER_GROUP2_BASE + 0x0005A000)
-#define EDMA1_CHANNEL_19_BASE_ADDR		(PER_GROUP2_BASE + 0x0005B000)
-#define EDMA1_CHANNEL_20_BASE_ADDR		(PER_GROUP2_BASE + 0x0005C000)
-#define EDMA1_CHANNEL_21_BASE_ADDR		(PER_GROUP2_BASE + 0x0005D000)
-#define EDMA1_CHANNEL_22_BASE_ADDR		(PER_GROUP2_BASE + 0x0005E000)
-#define EDMA1_CHANNEL_23_BASE_ADDR		(PER_GROUP2_BASE + 0x0005F000)
-#define EDMA1_CHANNEL_24_BASE_ADDR		(PER_GROUP2_BASE + 0x00060000)
-#define EDMA1_CHANNEL_25_BASE_ADDR		(PER_GROUP2_BASE + 0x00061000)
-#define EDMA1_CHANNEL_26_BASE_ADDR		(PER_GROUP2_BASE + 0x00062000)
-#define EDMA1_CHANNEL_27_BASE_ADDR		(PER_GROUP2_BASE + 0x00063000)
-#define EDMA1_CHANNEL_28_BASE_ADDR		(PER_GROUP2_BASE + 0x00064000)
-#define EDMA1_CHANNEL_29_BASE_ADDR		(PER_GROUP2_BASE + 0x00065000)
-#define EDMA1_CHANNEL_30_BASE_ADDR		(PER_GROUP2_BASE + 0x00066000)
-#define EDMA1_CHANNEL_31_BASE_ADDR		(PER_GROUP2_BASE + 0x00067000)
-#define PIT1_BASE_ADDR				(PER_GROUP2_BASE + 0x00088000)
-#define SEMA4_BASE_ADDR				(PER_GROUP2_BASE + 0x00098000)
-#ifdef CONFIG_NXP_S32R45
-#define CAN_FD2_BASE_ADDR			(PER_GROUP2_BASE + 0x000A8000)
-#define CAN_FD3_BASE_ADDR			(PER_GROUP2_BASE + 0x000B2000)
-#else
-#define CAN_FD2_BASE_ADDR			(PER_GROUP2_BASE + 0x000AC000)
-#define CAN_FD3_BASE_ADDR			(PER_GROUP2_BASE + 0x000B0000)
-#endif
-#define LINFLEXD2_BASE_ADDR			(PER_GROUP2_BASE + 0x000BC000)
-#define I2C4_BASE_ADDR				(PER_GROUP2_BASE + 0x000D8000)
-#define I2C5_BASE_ADDR				(PER_GROUP2_BASE + 0x000DC000)
-#define FLEXTIMER1_BASE_ADDR			(PER_GROUP2_BASE + 0x000E4000)
-#define SARADC1_BASE_ADDR			(PER_GROUP2_BASE + 0x000E8000)
-#define USDHC_BASE_ADDR				(PER_GROUP2_BASE + 0x000F0000)
-#define JDC_BASE_ADDR				(PER_GROUP2_BASE + 0x000F4000)
-#define FLEXRAY_BASE_ADDR			(PER_GROUP2_BASE + 0x000F8000)
-
-/* Peripheral group 3 */
-#define RCCU_BASE_ADDR				(PER_GROUP3_BASE)
-#define EIM_BASE_ADDR				(PER_GROUP3_BASE + 0x00008000)
-#define FCCU_BASE_ADDR				(PER_GROUP3_BASE + 0x0000C000)
-#define ETHERNET_0_BASE_ADDR			(PER_GROUP3_BASE + 0x0003C000)
-
-/* Peripheral group 8 */
-#ifdef CONFIG_NXP_S32R45
-#define SIUL2_1_BASE_ADDR			(PER_GROUP8_BASE + 0x0003c000)
-#define MC_CGM2_BASE_ADDR			(PER_GROUP8_BASE + 0x000c0000)
-#else
-#define SIUL2_1_BASE_ADDR			(PER_GROUP8_BASE + 0x00010000)
-#define MC_CGM2_BASE_ADDR			(PER_GROUP8_BASE + 0x00018000)
-#endif
-
-#endif	/* __ASM_ARCH_S32XXXX_GEN1_REGS_H__ */
diff --git a/drivers/i2c/mxc_i2c.c b/drivers/i2c/mxc_i2c.c
index b81c728515..077455f030 100644
--- a/drivers/i2c/mxc_i2c.c
+++ b/drivers/i2c/mxc_i2c.c
@@ -670,21 +670,11 @@ static int bus_i2c_write(struct mxc_i2c_bus *i2c_bus, u8 chip, u32 addr,
 	return ret;
 }
 
-#if !defined(I2C2_BASE_ADDR)
-#define I2C2_BASE_ADDR	0
-#endif
-
-#if !defined(I2C3_BASE_ADDR)
-#define I2C3_BASE_ADDR	0
-#endif
-
-#if !defined(I2C4_BASE_ADDR)
-#define I2C4_BASE_ADDR	0
-#endif
-
-#if !defined(I2C5_BASE_ADDR)
-#define I2C5_BASE_ADDR 0
-#endif
+#define I2C1_BASE_ADDR		(0x401E4000)
+#define I2C2_BASE_ADDR		(0x401E8000)
+#define I2C3_BASE_ADDR		(0x401EC000)
+#define I2C4_BASE_ADDR		(0x402D8000)
+#define I2C5_BASE_ADDR		(0x402DC000)
 
 #if !defined(I2C6_BASE_ADDR)
 #define I2C6_BASE_ADDR 0
diff --git a/drivers/pci/serdes_s32gen1.h b/drivers/pci/serdes_s32gen1.h
index 29dae1c04f..e2a8cea4dc 100644
--- a/drivers/pci/serdes_s32gen1.h
+++ b/drivers/pci/serdes_s32gen1.h
@@ -9,7 +9,6 @@
 
 #include <pci.h>
 #include <dm.h>
-#include <asm/arch/s32-gen1/s32-gen1-regs.h>
 #include <asm/io.h>
 
 #include "serdes_regs.h"
@@ -45,6 +44,7 @@
 #define COLD_RST		0x1
 
 /* RGM peripheral reset registers */
+#define UPTR(PTR)			((uintptr_t)(PTR))
 #define RGM_PRST(MC_RGM, per)		(UPTR(MC_RGM) + 0x40 + \
 					((per) * 0x8))
 #define RGM_PSTAT(rgm, per)		(UPTR(rgm) + 0x140 + \
@@ -56,6 +56,8 @@
 #define PRST_PCIE_1_SERDES		16
 #define PRST_PCIE_1_FUNC		17
 
+#define MC_RGM_BASE_ADDR		(0x40078000)
+
 enum serdes_link_width {
 	X1 = 0x1,
 	X2 = 0x2
-- 
2.25.1

